include/boost/histogram/detail/probit.hpp

```diff
@@ -0,0 +1,37 @@
+// Copyright 2022 Jay Gohil, Hans Dembinski
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_HISTOGRAM_DETAIL_PROBIT_HPP
+#define BOOST_HISTOGRAM_DETAIL_PROBIT_HPP
+
+#include <cmath>
+
+namespace boost {
+namespace histogram {
+namespace detail {
+
+template <class T>
+T erf_inv_approx(T x){
+   T one = T(1);
+   T lnx = std::log((one - x)*(one + x));
+
+   T tt1 = T(4.120666747961526) + T(0.5) * lnx;
+   T tt2 = T(6.47272819164) * lnx;
+
+   return std::copysign(std::sqrt(-tt1 + std::sqrt(tt1*tt1 - tt2)), x);
+}
```

Comment:
I think it is sufficient to have erf_inv_approx in double precision, and I also think that the `approx` should be in front, so please replace this with this slightly more optimized version
```cpp
    inline double approx_erf_inv(double x) {
        const double lnx = std::log((1 - x)*(1 + x));
        const double tt1 = std::fma(0.5, lnx, 4.120666747961526);
        const double tt2 = 6.47272819164 * lnx;
        return std::copysign(std::sqrt(-tt1 + std::sqrt(std::fma(tt1, tt1, -tt2))), x);
    }
```
`inline` is needed when you define a function inside a header and it is not templated.

Reply:
std::fma does the operation a * x + b in one CPU cycle. We need to use this explicitly here, because the compiler cannot optimize a * x + b into the fma instruction automatically unless special compiler flags are used (which are usually not turned on). The optimizer is not allowed to perform code transformations that could produce a different result.

Reply:
You can ignore this request, because I made the change in my commit to your PR. I replaced detail/probit.hpp with detail/normal.hpp and improved the erf_inv implementation. I still leave the comments here for you to read.